services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/ProtectedPathSpec.groovy

```diff
@@ -0,0 +1,942 @@
+        given: "Two active neighboring switches"
+        def isls = topology.getIslsForActiveSwitches()
+        def (srcSwitch, dstSwitch) = [isls.first().srcSwitch, isls.first().dstSwitch]
+
+        and: "Update all isls which can be used by protected path"
+        def bandwidth = 100
+        isls[1..-1].each { isl ->
+            database.updateIslAvailableBandwidth(isl, 90)
+            database.updateIslAvailableBandwidth(isl.reversed, 90)
+        }
+
+        when: "Create flow without protected path"
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow.maximumBandwidth = bandwidth
+        flow.allocateProtectedPath = false
+        flow.ignoreBandwidth = true
+        flowHelper.addFlow(flow)
+
+        then: "Flow is created without protected path"
+        !northbound.getFlowPath(flow.id).protectedPath
+
+        when: "Update flow: enable protected path(allocateProtectedPath=true)"
+        northbound.updateFlow(flow.id, flow.tap { it.allocateProtectedPath = true })
+
+        then: "Protected path is enabled"
+        northbound.getFlowPath(flow.id).protectedPath
+
+        // TODO I've second thoughts about the following lines
+        northbound.updateFlow(flow.id, flow.tap { it.ignoreBandwidth = false })
+        northbound.updateFlow(flow.id, flow.tap { it.allocateProtectedPath = false })
+        //Should the protected path be enabled?
+//        isls[1..-1].each { isl ->
+//            database.updateIslAvailableBandwidth(isl, 90)
+//            database.updateIslAvailableBandwidth(isl.reversed, 90)
+//        }
+        northbound.updateFlow(flow.id, flow.tap { it.allocateProtectedPath = true })
+
+        and: "Cleanup: delete the flow and restore available bandwidth"
+        flowHelper.deleteFlow(flow.id)
+        isls.each { isl ->
+            database.resetIslBandwidth(isl)
+            database.resetIslBandwidth(isl.reversed)
+        }
+    }
+
+    def "System is able to recalculate protected path when port is down"() {
+        given: "Two active not neighboring switches with two possible paths at least"
+        def allIsls = northbound.getAllLinks()
+        def (Switch srcSwitch, Switch dstSwitch) = getNotNeighboringSwitchPair(2)
+
+        when: "Create a flow with protected path"
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow.allocateProtectedPath = true
+        flowHelper.addFlow(flow)
+
+        then: "Flow is created with protected path"
+        def flowPathInfo = northbound.getFlowPath(flow.id)
+        flowPathInfo.protectedPath
+
+        and: "Current path is not equal to protected path"
+        def currentPath = pathHelper.convert(flowPathInfo)
+        def currentProtectedPath = pathHelper.convert(flowPathInfo.protectedPath)
+        currentPath != currentProtectedPath
+
+        and: "Current path is not changed"
+        currentPath == pathHelper.convert(northbound.getFlowPath(flow.id))
```

Comment:
Why you check this?

Reply:
Roman ask me to do it

Reply:
Ok, for what is the goal of this?

Reply:
My though process was to ensure that main path does not get rerouted if reroute is triggered on protected path.

Reply:
1. We just created a flow. No events were made to trigger reroute.
2. I have the same vision. So this check should do after verifying that the protected path is recalculated (L548) but not here.

Reply:
Agree. Didn't see the exact place where this verification was put

Reply:
My fault
this check should be put after  then `the: "Protected path is recalculated"` block